platform/openide.util/test/unit/src/org/openide/util/BaseUtilitiesTest.java

```diff
@@ -111,31 +112,106 @@ public void testIsJavaIdentifier() throws Exception {
-        } else {
-            assertFileURI("/some/path #1", "file:/some/path%20%231");
-            assertEquals(new File("/some/path"), BaseUtilities.toFile(new URI("file:/some/path")));
-            assertEquals(new File("/some/path"), BaseUtilities.toFile(new URI("file:///some/path")));
-            assertEquals(new File("/some/path"), BaseUtilities.toFile(new URI("file:/some/path/")));
+    //--------------------------------------------------------------------------
+    public void test_toFile_throwsNullPonter_whenArgumentIsNull()
+            throws Exception {
+
+        try {
+            toFile(null);
+            fail();
+        } catch (final NullPointerException e) {
+            return;
+        }
+    }
+    
+    //--------------------------------------------------------------------------
+    public void test_toFile_throwsIllegalArgument_whenGivenNonFileURI()
+            throws Exception {
+
+        try {
+            toFile(new URI("http://example.com"));
+            fail();
+        } catch (final IllegalArgumentException e) {
+            return;
+        }
+        try {
+            toFile(new URI("mailto:person@example.com"));
+            fail();
+        } catch (final IllegalArgumentException e) {
+            return;
+        }
+        try {
+            toFile(new URI("isbn:12345"));
+            fail();
+        } catch (final IllegalArgumentException e) {
+            return;
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    public void test_toFile_returnsFile_whenGivenUnixPath()
+            throws Exception {
+
+        assertEquals(new File("/some/path"), toFile(new URI("file:/some/path")));
+        assertEquals(new File("/some/path"), toFile(new URI("file:///some/path")));
+        assertEquals(new File("/some/path"), toFile(new URI("file:/some/path/")));
+        assertEquals(new File("/some/path #1"), toFile(new URI("file:/some/path%20%231")));
+    }
+
+    //--------------------------------------------------------------------------
+    public void test_toFile_returnsFile_whenGivenWindowsPath()
+            throws Exception {
+
+        if (isWindows()) {
+            assertEquals(new File("C:\\some\\path #1"), toFile(new URI("file:/C:/some/path%20%231")));
+            assertEquals(new File("C:\\some\\path"), toFile(new URI("file:/C:/some/path")));
+            assertEquals(new File("C:\\some\\path"), toFile(new URI("file:///C:/some/path")));
+            assertEquals(new File("C:\\some\\path"), toFile(new URI("file:/C:/some/path/")));
+            assertEquals(new File("\\\\server\\share\\path"), toFile(new URI("file://server/share/path")));
+            assertEquals(new File("\\\\server\\share\\path"), toFile(new URI("file:////server/share/path")));
+            assertEquals(new File("\\\\server\\share\\path #1"), toFile(new URI("file:////server/share/path%20%231")));
+        }
+    }
```

Comment:
me neither - but iexecuting toFile(new URI("file:/some/path")) under windows gives "d:\some\path" on my machine

I have no idea how to work arround this right now - the original test had the same issue

Reply:
I see - so there is a valid reason for the os checks. So lets merge the os - dependent cases into os - independent cases with an if branch in them like it was before to avoid green no-ops.

Reply:
thanks!